resources/intro-to-observables.md

```diff
@@ -0,0 +1,123 @@
+
+For example, we often see this in the context of API calls when we request data from another server or backend service. We make a request for the data on the side and, once the data arrives, we put it in a queue to be processed. We don't actually stop and hold up all activity in the app until the data comes back. Instead, we put it to the side and process it when it arrives (and when we're ready) - [*insert link to event loop bridge content here*].
+
+We also encounter this with DOM events, timer intervals, web workers and web sockets.
+
+**Don't we already have a few ways to handle asynchronous data?** 
+Yes, we can use callback functions and promises for example. But Observables are slightly different as they allow us to process **streams of data**.
+
+## Why should I care?
+
+Observables play a key part in something called **reactive programming**. This is the concept that anything can be turned into a data stream (a sequence of values in time). You can choose to listen to these streams and, as data comes through them, react accordingly. 
+
+Data streams are functional - so you can use streams as inputs into other streams, combine them, filter them, etc using a variety of helpful functions. 
+
+Observables are now included in the current version of JavaScript (as of ES7) and Angular now has them baked in using a third-party library called RxJS (Reactive Extensions). In Angular, Observables are used for its event system and HTTP client service. In addition, we're starting to see more React projects use RxJS to take advantage of Observables to manage streams.
+
+## What is an Observable?
+
+Observables originate from something called the **observer design pattern**. 
+
+> The observer pattern is a software design pattern in which an object, called the **subject**, maintains a list of its dependents, called **observers**, and notifies them automatically of any state changes, usually by calling one of their methods. ([source](https://en.wikipedia.org/wiki/Observer_pattern)) 
+
+Ok, so that definition is pretty wordy and technical. 
+
+In simple terms, the pattern involves an object (called an observable) that will share (or emit) data as it arrives over time. Throughout your code, many different objects (called observers) can sign up to be notified when that data arrives (called subscribing). So, whenever our observable recieves its data, it will broadcast that information out and the observers will be listening for it.  
+
+Some key points to note about the pattern:
+
+- You can have one observable and many observers (a one-to-many dependancy). That is, one broadcaster and an unlimited number of recipients.
+- When that observable changes, all observers should be updated/notified immediately and automatically
+- It is the observer's responsibility to register and unregister themselves from the observable (to be notified of changes) and to update their own state when notified.
+
```

Comment:
Would it be helpful to have a separate example of something using the observer pattern here? Or just go straight into observables in the next section?

Reply:
Go straight in, any 'non observable' examples can be in a separate file if need be

Reply:
👍